src/streams/StreamList.js

```diff
@@ -77,8 +77,14 @@ export default class StreamList extends PureComponent {
             showSwitch={showSwitch}
             isSwitchedOn={item.subscribed}
             onPress={onPress}
-            onSwitch={onSwitch}
-          />}
+            onSwitch={
```

Comment:
We do not inline function declarations inside a component like this.
The reason is that a new function is created for each component render. This not only wasted memory but also forces a repaint of the item, even when no other props have changed.

In this case, it also creates multiple functions per render as there is a list of StreamItems being rendered.

The solution is a `handleSwitch` method and then you do `onSwitch={this.handleSwitch}`